### ISSUE TITLE
R4i Gold 3ds (r4ids.cn)

### ISSUE THREAD

**Issue by Banamy** [kitling/flashcart_core#9]
Alright, I got some pics for you all with my r4 gold 3ds. This is the LAST update that officially worked with this revision. You can find the revision on the bottom right on the back of the card, on the board.
http://filetrip.net/dl?9l07ieOpEm - .nds update file

Pics arent sideways if you click on them, sorry bout that.

![20170813_083707](https://user-images.githubusercontent.com/18485292/29256103-a74e227c-8075-11e7-86a7-e1c46efd075f.jpg)
![20170813_083714](https://user-images.githubusercontent.com/18485292/29256102-a74c9e02-8075-11e7-95af-903f8462d9a3.jpg)
![20170813_083747](https://user-images.githubusercontent.com/18485292/29256105-a75059de-8075-11e7-8354-8a6b2bcd5afc.jpg)
![20170813_083811](https://user-images.githubusercontent.com/18485292/29256104-a74e6462-8075-11e7-8a5d-598344eb7493.jpg)
![20170813_085137](https://user-images.githubusercontent.com/18485292/29256106-a7531ed0-8075-11e7-8c46-11919e4d51fb.jpg)
![20170813_085315](https://user-images.githubusercontent.com/18485292/29256107-a92e2600-8075-11e7-8449-383251aa635b.jpg)
![sd card](https://user-images.githubusercontent.com/18485292/29256108-a9306672-8075-11e7-91a6-43176b34a691.PNG)

**Comment by dark-samus** [comment]
added to the todo list :)

**Comment by mariogamer2** [comment]
I've got a similar flashcard from r4i-sdhc.com, but I don't think it's compatible... It's also use the main menu to upgrade the flashcards... Should I create another issue?

**Comment by pixelmasher** [comment]
@dark-samus I have the same card, but revision C. That one has HW Rev: A7A7A7A7. Shows up as R4i Gold 3DS. Works fine with ntrboot flasher 0.1.2.

**Comment by AnalogMan151** [comment]
I have revision D which also works with NTRBootHax.

**Comment by Banamy** [comment]
Revision 6 doesnt :/

**Comment by AnalogMan151** [comment]
Obviously.

**Comment by Banamy** [comment]
Pretty sure any card before the cutoff doesnt work. So all pre 4.5 r4i gold 3ds, dont work.

**Comment by Juanito9001** [comment]
Somebody knows if the r4igold from "r4igold.cn" works with ntrboot?

_Sent from my Samsung SM-G610M using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by kitling** [comment]
@Juanito9001, if this is the cart you're talking about mentioned in the issue... well, it's been over what they think works and doesn't work at the moment.

If it's not this card mentioned in this issue, then this is *not the place*. If you want to ask questions like this, I'd suggest that IRC would be a better place (though the answer is most likely to be "No", "Not Yet", or "We don't know").

**Comment by coopmaster123** [comment]
so did this end up being supported or work? or did it brick your card?

**Comment by dark-samus** [comment]
@Banamy when you're around, please attempt to dump flash with this: 
[ntrboot_flasher.zip](https://github.com/kitling/flashcart_core/files/1253164/ntrboot_flasher.zip)

post the result here. An (untrimmed) dump from GM9 would be useful as well :)

**Comment by Banamy** [comment]
http://s000.tinyupload.com/index.php?file_id=02820229957007339507 - dump flash

https://mega.nz/#!qkUnQRCZ!iPRJ1S1ksmggytkI_AuuxgJLOPOAWHg_Q_V5gmBDdsg - untrimmed dump

**Comment by dark-samus** [comment]
excellent, it looks like the flash is dumpable with the standard commands :D it does have different crypto though. I'll see what I can do :)

**Comment by dark-samus** [comment]
I see why I didn't just RE this thing at first; I was busy with another flashcart and most of the program seems encrypted. Multiple layers too. I've already got through one of the layers, but it appears that they actually open the .nds file again and do some manipulation there... It's odd but I'm getting there...

**Comment by shinji257** [comment]
Just wanted to throw my name in the hat for possible testers down the road when you are ready.  I did note one thing when making my own dump.  It showed all 0's when there was no microSD inserted.  It showed FC2 as above when one was present.  The dump itself didn't change regardless of the state.  Just thought to let you know.

**Comment by Banamy** [comment]
I get the same thing

**Comment by dark-samus** [comment]
does HW Rev change as well?

**Comment by Banamy** [comment]
HW Rev is always all 0

**Comment by dark-samus** [comment]
lol, ok, looks like the encryption for this revision is the same as the other revs. The updater checks for HW rev 0xA4A4A4A4, so I'm not sure why we're getting that HW rev.... in any case, it should be fairly simple to support your cart once we figure that out :)

**Comment by dark-samus** [comment]
tell me what HW rev this build says when you try to dump flash.
[ntrboot_flasher.zip](https://github.com/kitling/flashcart_core/files/1253355/ntrboot_flasher.zip)

**Comment by Banamy** [comment]
00000000

**Comment by dark-samus** [comment]
and this? 
[ntrboot_flasher.zip](https://github.com/kitling/flashcart_core/files/1253387/ntrboot_flasher.zip)

**Comment by shinji257** [comment]
Still showing as 00000000

**Comment by Banamy** [comment]
same

**Comment by zoogie** [comment]
@dark-samus I tried to use the same approach that worked with my rev5 with Banamy's rev6, but it didn't work for him. (maybe the blowfish goes somewhere other than 0?)

Anyway, here's my notes in case they may help:
ntr_blowfish_retail = 0x0 (plain)
b9s_payload = 0x80000 (enc)
b9s_header = 0x1FFE00 (enc)
enc = same memecrypto as latest r4ids.cn versions

**Comment by dark-samus** [comment]
the main problem really is detection. Though, that may be going away in the near future...

**Comment by PrecastCrane02** [comment]
I got one of these carts too, is there anything I can do to help?

**Comment by PrecastCrane02** [comment]
It worked @dark-samus. Cool. Is there anything else I could do to help?

**Comment by PrecastCrane02** [comment]
@dark-samus I might just be doing the whole thing wrong, but when I use that build everything seems fine. I dump the flash and after the dump I use the inject ntrboot option. However, I do not get booted into b9safeboot on the 3ds I'm trying to hack (N3DS)(I do have the boot files etc.). Am I using this wrong? Or is my cart broken? (Also, I should not have a SD-card in the flashcart slot, right?)

**Comment by dark-samus** [comment]
send me the flash dump please

**Comment by PrecastCrane02** [comment]
I'm sorry, but how do I send you the dump? Should I just use the flasher? And where can I find the dump?

**Comment by PrecastCrane02** [comment]
I got the backup.bin here http://s000.tinyupload.com/index.php?file_id=04402297391933793406

**Comment by Inversil** [comment]
Any updates on this?

**Comment by Leadger69** [comment]
I got the same problem as precastcrane02 has ..it dump and inject but don't load safeboot

**Comment by zoogie** [comment]
https://gbatemp.net/threads/r4igold-r4ids-cn-rev-6-7-8-offering-a-2ds-ntrboot-ak2i-or-20-eshop.488295/

I'm offering a 2ds for one of the cards (rev 6-8). Only way to get this thing supported. Desperate. XD

**Comment by PrecastCrane02** [comment]
@zoogie Got one for ya, I would love to trade one for a working NTRboot flashed cart, instead of an 2DS however.

**Comment by zoogie** [comment]
@PrecastCrane02 - Thanks a bunch for the kind offer, but my 2ds is already off to someone. :/ 
However, in case my current trade falls through, I might take you up on that.

**Comment by zoogie** [comment]
Got it figured out. The blowfish location was at 0x1FA000 instead of 0 and it's encrypted.
https://gist.github.com/zoogie/bc4edb08c8e9fce6276b22562424f9b0

**Comment by d3m3vilurr** [comment]
anyone knows C7 command returned values of rev 4~7?

**Comment by zoogie** [comment]
@d3m3vilurr 
How would I go about checking that c7 command?

**Comment by zoogie** [comment]
@Leadger69 
Could you do me (and @d3m3vilurr) a favor?
Using this
[ntrboot_flasher.zip](https://github.com/ntrteam/flashcart_core/files/1472600/ntrboot_flasher.zip)
Please insert your rev7, boot the app, press Y until you see "Log Level: DEBUG" in the bottom right of the top screen. Then "Dump Flash". The backup.bin dump isn't important, look at ntrboot.log in sdmc://ntrboot/ and tell us what number is after "[NOTICE]: R4iGold: HW C7 value = "

That would help a bunch in getting your card supported.

**Comment by d3m3vilurr** [comment]
@zoogie interesting. cuz my A7 also return `b7db5bb5`

anyway, maybe we can split two big groups
- `a79bca95`, `9bca95a7`, `ca95a79b`, `95a79bca`
- `b7db5bb5`, `db5bb5b7`, `5bb5b7db`, `b5b7db5b`

need more verification

**Comment by Leadger69** [comment]
I just had to back up and reformat my pc so I'll get this done for you guys later on tonight.
Thanks for all the help by the way

**Comment by PrecastCrane02** [comment]
@zoogie Can I help in any way? (Rev. 6)

**Comment by zoogie** [comment]
@PrecastCrane02 
Yeah, first do the steps in https://github.com/ntrteam/flashcart_core/issues/9#issuecomment-344408217
Then post the results here please. 
After that I might have a beta for you to try. : )

**Comment by PrecastCrane02** [comment]
@zoogie I got a HW C7 value of "b7db5bb5"

**Comment by zoogie** [comment]
@PrecastCrane02 
Great! Thank you.

If you want, go ahead and try this little beta I made and see if ntrboothax works. Make sure to create a backup first (Dump Flash).
https://github.com/zoogie/flashcart_core/releases

**Comment by PrecastCrane02** [comment]
@zoogie Amazing work! It works! Thank you guys so much.

**Comment by Leadger69** [comment]
Yeah my HW  C7 value is b7db5bb5

**Comment by zoogie** [comment]
ok thanks @Leadger69 !
Do you want to try the beta now?

Look at https://github.com/ntrteam/flashcart_core/issues/9#issuecomment-344724459 and do those steps.

**Comment by Leadger69** [comment]
It worked thanks so much

**Comment by shlomi-toren-sp** [comment]
Is it supposed to take hours? It's been almost three hours and Im on chunk 4. I may be on debug.

**Comment by kitling** [comment]
If it's progressing, just really slowly... then it should be okay? I'm sorry that debug logging is so slow.

If, on the other hand you got to chunk 4 quickly and now it isn't progressing... then there may be an actual issue.

**Comment by shlomi-toren-sp** [comment]
@kitling  It was going forward consistently. Took 7 hours, but finally it finished and everything works now, so it's fat from crucial.
I'm planning to flash it again anyway, so I'll try without debug next time, just to let you know about the time difference.

**Comment by kitling** [comment]
yeah, debug logging is just really slow, since we mount/unmount the SD card filesystem every time we log a single line.
